File: context_os_runtime/runtime_paths.py
from __future__ import annotations

from pathlib import Path


def runtime_dir(repo_root: Path) -> Path:
    return repo_root / ".agent-os" / "runtime"


def event_log_path(repo_root: Path) -> Path:
    return runtime_dir(repo_root) / "events.jsonl"


def session_snapshot_path(repo_root: Path) -> Path:
    return runtime_dir(repo_root) / "session.json"

File: src/core/runtime-paths.ts
import { join } from 'node:path';

export function runtimeDir(repoRoot: string): string {
  return join(repoRoot, '.agent-os', 'runtime');
}

export function lockPath(repoRoot: string): string {
  return join(repoRoot, '.agent-os.lock');
}

export function eventLogPath(repoRoot: string): string {
  return join(runtimeDir(repoRoot), 'events.jsonl');
}

export function sessionSnapshotPath(repoRoot: string): string {
  return join(runtimeDir(repoRoot), 'session.json');
}

export function sessionDir(repoRoot: string, sessionId: string): string {
  return join(runtimeDir(repoRoot), 'sessions', sessionId);
}

export function sessionEventsPath(repoRoot: string, sessionId: string): string {
  return join(sessionDir(repoRoot, sessionId), 'events.jsonl');
}

export function sessionDashboardPath(repoRoot: string, sessionId: string): string {
  return join(sessionDir(repoRoot, sessionId), 'dashboard.json');
}
